### PR TITLE
Convert dotnet/runtime to use the new Arcade tooling for RPMs

### LINKED ISSUE #4138: Runtime builds not generating deb or rpm installers
The runtime builds on Linux are not generating the .deb installers required for the full .NET SDK build.

### LINKED ISSUE #109518: Move dotnet/runtime to new RPM tooling
dotnet/arcade has new tooling to build RPMs that does not require FPM. We need to move to the new RPM tooling to enable building on the same docker image for all steps. Additionally, moving to the new tooling will further enable us to build all of our linux packages as part of the regular build (and no longer require a second build invocation).

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #109518
Fixes https://github.com/dotnet/source-build/issues/4138
Contributes to https://github.com/dotnet/source-build/issues/3986

Depends on https://github.com/dotnet/arcade/pull/15217

### CODE REVIEW

File: src/installer/pkg/sfx/installers/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props, $(MSBuildThisFileDirectory)..))" />
+  
+  <PropertyGroup>
+    <BuildDebPackage Condition="'$(TargetsLinuxGlibc)' == 'true'">true</BuildDebPackage>
+    <BuildRpmPackage Condition="'$(TargetsLinuxGlibc)' == 'true'">true</BuildRpmPackage>
```
Comment (am11) [inline src/installer/pkg/sfx/installers/Directory.Build.props:6]: These are set in .proj files, do we also need them in .props file?

Comment (jkoritzinsky) [reply src/installer/pkg/sfx/installers/Directory.Build.props:6]: Yes. This provides the defaults for the installers, but the runtime-deps installers only build some of the flavors (deb or rpm, not both).

**Comment by jkoritzinsky** [comment]
Looks like cpio isn't available in the Azl3 docker images we use... I might have to update the image to install it.

**Comment by jkoritzinsky** [comment]
Apparently `file` isn't included either in the base image and the source-build images don't have `cpio`.

Docker image updates in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1247

**Review by ViktorHofer — APPROVED** [review]
Amazing, nice diff! Especially the YML part.

**Pull request merged**